applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationClientTest.java

```diff
@@ -642,24 +643,32 @@ private static void assertConfigurationEquals(ConfigurationSetting expected, Con
         }
     }
 
+
+    /**
+     * Helper method to verify the runner throws a HttpRequestException and it has a specific HTTP response code.
+     *
+     * @param exceptionThrower Runnable that throws the exception.
+     * @param expectedStatusCode Expected HTTP status code contained in the error response
+     */
     private static void assertRestException(Runnable exceptionThrower, int expectedStatusCode) {
-        try {
-            exceptionThrower.run();
-            fail();
-        } catch (Throwable ex) {
-            assertRestException(ex, expectedStatusCode);
-        }
+        assertRestException(exceptionThrower, HttpRequestException.class, expectedStatusCode);
     }
 
     /**
-     * Helper method to verify the error was a RestException and it has a specific HTTP response code.
+     * Helper method to verify the runner throws the expected exception type and it has a specific HTTP response code.
      *
-     * @param ex Expected error thrown during the test
+     * @param exceptionThrower Runnable that throws the exception.
+     * @param expectedExceptionType Expected exception type returned.
      * @param expectedStatusCode Expected HTTP status code contained in the error response
      */
-    private static void assertRestException(Throwable ex, int expectedStatusCode) {
-        assertTrue(ex instanceof ServiceRequestException);
-        assertEquals(expectedStatusCode, ((HttpRequestException) ex).response().statusCode());
+    private static void assertRestException(Runnable exceptionThrower, Class expectedExceptionType, int expectedStatusCode) {
+        try {
+            exceptionThrower.run();
+            fail();
+        } catch (Throwable ex) {
+            assertEquals(ex.getClass(), expectedExceptionType);
+            assertEquals(expectedStatusCode, ((HttpRequestException) ex).response().statusCode());
```

Comment:
Is it possible to constrain the Class expectedExceptionType to be assignable to HttpRequestException? 🤔
You don't have to change it if it requires too much work. I was wondering about the cast here failing with an InvalidCastException if `ex` could not be casted to `HttpRequestException`.

Reply:
Done